examples/hotrod/pkg/log/factory.go

```diff
@@ -53,7 +53,14 @@ func (b Factory) For(ctx context.Context) Logger {
 				zap.String("span_id", otelSpan.SpanContext().SpanID().String()),
 			}
 		}
-
+		return logger
+	}
+	if otelSpan := trace.SpanFromContext(ctx); otelSpan != nil {
```

Comment:
A hack to fallback to trying to fetch the OTEL span if trying to fetch the OT span fails.

Reply:
Thanks @albertteoh to hacking around this. Supposedly, I thought we needed this. You made it. Awesome! 💯